modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

```diff
@@ -0,0 +1,79 @@
+:_content-type: PROCEDURE
+
+[id="adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance"]
+= Adding or removing pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance
+
+You can add or remove your choice of pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance that is cached by {prod-short} to support air-gapped, offline, and proxy-restricted environments.
+
+This will create a custom build of the Open VSX registry, which can be used in your organization’s workspaces. 
+
+Note that after a subscription update, your CheCluster will revert to default settings and any changes will need to be reapplied. 
+
+.Procedure
+
+include::example$snip_{project-context}-steps-to-add-or-remove-extensions-in-openvsx-sync-json.adoc[]
+
+** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add it by including an object with the extension `id` in the format __<published_by>__.__<unique_identifier>__ that you can collect from the extension page on link:https://open-vsx.org[open-vsx.org]:
++
+[source,json,subs="+quotes"]
+----
+    {
+        "id": "__<published_by>__.__<unique_identifier>__"
+    }
+----
++
+[TIP]
+====
+The latest extension version on link:https://open-vsx.org[open-vsx.org] is the default. Alternatively, you can add `"version": "__<extension_version>__"` on a new line to specify a version.
+====
+
+** If it is an extension from link:https://marketplace.visualstudio.com/VSCode[Visual Studio Marketplace] that is not available on link:https://open-vsx.org[Open VSX] yet, you can ask the extension author to publish it on link:https://open-vsx.org[open-vsx.org] according to the link:https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension[instructions] so that you can add it as an extension that is published on link:https://open-vsx.org[open-vsx.org].
++
+[TIP]
+====
+If the extension author is not cooperative, you can add the extension directly from the `.vsix` file, as long as the license and terms of use allow it, and the file can be accessed from a https:// URL while building the plugin registry container.
```

Comment:
@nickboldt and @svor, the only two places where I could find `.vsix` files for download were https://marketplace.visualstudio.com/ and https://open-vsx.org/, and neither of them provides clear URLs that I could use for downloading `.vsix` files.

So I wonder if I am missing some other use case that does not violate MIcrosoft's TOU. If not, then maybe we should indeed remove this TIP as Nick suggested previously? If no objections, then I propose replacing this TIP with the TIP in Nick's previous comment ^. WDYT?

Reply:
There are 4 user stories:

* extension is already on Open VSX: **OK to use**

* extension is on MS VSC marketplace: **NEVER OK to use**. See TOU. 

* I (on behalf of my company) built my own extension, but didn't publish it to a public site because I'm into closed source or it's got prioprietary stuff for my company, so it's on an internal company website from which it can be fetched into the registry container build: not great Open Source ethics, but as long as the license is respected, **this is valid**

* I want something from MS VSC marketplace so I reached out to the publisher and they agreed to publish it to Open VSX: as long as the publisher provided it to Open VSX and we use it from THAT URL... **OK to use**. Perhaps we link to https://github.com/marketplace/actions/publish-vs-code-extension to make it super easy? 

>  some other use case that does not violate Microsoft's TOU

Please read the TOU. It's pretty concise in that you can ONLY use the marketplace w/ MS products. Period. End of story.

Reply:
OK, thank you, Nick. Let me have one more try at it.